lib/phlex/kit.rb

```diff
@@ -1,31 +1,39 @@
 # frozen_string_literal: true
 
 module Phlex::Kit
-	def self.extended(mod)
-		warn "⚠️ [WARNING] Phlex::Kit is experimental and may be removed from future versions of Phlex."
-		super
+	module LazyLoader
+		def method_missing(name, *args, **kwargs, &block)
+			if name[0] == name[0].upcase && __phlex_kit_constants__.include?(name) && __get_phlex_kit_constant__(name) && methods.include?(name)
+				public_send(name, *args, **kwargs, &block)
+			else
+				super
+			end
+		end
+
+		def respond_to_missing?(name, include_private = false)
+			if name[0] == name[0].upcase && __phlex_kit_constants__.include?(name) && __get_phlex_kit_constant__(name) && methods.include?(name)
+				true
+			else
+				super
+			end
+		end
```

Comment:
This module hooks into `method_missing` and `respond_to_missing?` to lazy-load an *autoloaded* constant with the same name as the method.

The first check `name[0] == name[0].upcase` is for performance as it allows it to exist early in the case that the missing method doesn’t even start with a capital letter.

Next, it checks if a constant matching the method name exists. It goes through `__phlex_kit_constants__`, since this is sometimes called on a class and sometimes on an instance, and they both need to handle this call.

If a matching constant is found, it then gets the constant via `__get_phlex_kit_constant__`. This will trigger an autoloaded constant to be loaded. If this constant is indeed a `Phlex::SGML` component, loading it will automatically define the missing method that we’re responding to so it won’t be missing anymore.

Finally, we check if that method was defined with `methods.include?(name)`. If it was, we can call it again. It won’t cause an infinite loop because this time, we won’t end up in the `method_missing` function.